dojo_theme/templates/workspace_port.html

```diff
@@ -0,0 +1,72 @@
+{% extends "base.html" %} {% block stylesheets %} {{ super() }}
+<style>
+    html,
+    body {
+        margin: 0;
+        height: 100%;
+        min-height: 100%;
+    }
+    body {
+        display: flex;
+        flex-direction: column;
+    }
+    nav,
+    footer {
+        flex: none;
+    }
+    main {
+        flex: auto;
+        margin-bottom: 0px;
+    }
+    iframe {
+        height: 100%;
+        width: 100%;
+        overflow: hidden;
+        border: none;
+    }
+    .inactive {
+        height: 50%;
+        display: flex;
+        flex-direction: column;
+        justify-content: center;
+        text-align: center;
+    }
+</style>
+{% endblock %} {% block content %}
+
+<iframe
+    id="workspace_iframe"
+    name="{{ iframe_name }}"
+    src=""
+    class="d-none"
+></iframe>
+<h1 class="inactive">Loading...</h1>
+
+{% endblock %} {% block scripts %} {{ super() }}
+
+<script>
+    let response = fetch("/pwncollege_api/v1/workspace?port={{ port }}")
```

Comment:
Realistically this template could be merged with workspace_service.html, since `?port=` vs `?service=` is the only difference.